components/num-util/src/uint_iterator.rs

```diff
@@ -0,0 +1,102 @@
+use std::convert::TryInto;
+
+/// An iterator over the decimal digits of an integer, from least to most significant
+pub(crate) struct IntIterator<T> {
+    /// Digits remaining to be returned from the iterator
+    unum: T,
+    /// Whether the number is negative
+    pub is_negative: bool,
+}
+
+macro_rules! impl_from_signed_integer_type {
+    ($itype:ident, $utype:ident) => {
+        impl From<$itype> for IntIterator<$utype> {
+            fn from(value: $itype) -> Self {
+                IntIterator {
+                    unum: {
+                        if value == std::$itype::MIN {
+                            std::$itype::MAX as $utype + 1
+                        } else {
+                            value.abs() as $utype
+                        }
+                    },
+                    is_negative: value.is_negative(),
+                }
+            }
+        }
+    };
+}
+
+macro_rules! impl_from_unsigned_integer_type {
+    ($utype:ident) => {
+        impl From<$utype> for IntIterator<$utype> {
+            fn from(value: $utype) -> Self {
+                IntIterator {
+                    unum: value,
+                    is_negative: false,
+                }
+            }
+        }
+    };
+}
+
+impl_from_signed_integer_type!(isize, usize);
+impl_from_signed_integer_type!(i128, u128);
+impl_from_signed_integer_type!(i64, u64);
+impl_from_signed_integer_type!(i32, u32);
+impl_from_signed_integer_type!(i16, u16);
+impl_from_signed_integer_type!(i8, u8);
+
+impl_from_unsigned_integer_type!(usize);
+impl_from_unsigned_integer_type!(u128);
+impl_from_unsigned_integer_type!(u64);
+impl_from_unsigned_integer_type!(u32);
+impl_from_unsigned_integer_type!(u16);
+impl_from_unsigned_integer_type!(u8);
+
+impl<T> Iterator for IntIterator<T>
```

Comment:
@zbraniecki Should I stick this part into the `macro_rules!` as well, instead of using all these traits?

Reply:
I decided to do it.